service/src/main/scala/za/co/absa/logingw/rest/config/GitConfig.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.logingw.rest.config
+
+import org.springframework.boot.context.properties.{ConfigurationProperties, ConstructorBinding}
+import javax.annotation.PostConstruct
+import java.io.PrintWriter
+import java.util.Date
+import java.text.SimpleDateFormat
+
+@ConstructorBinding
+@ConfigurationProperties(prefix = "logingw.rest.config")
+case class GitConfig(gitProperties: Boolean) {
+
+  @PostConstruct
+  def init(): Unit = {
+    if(gitProperties)
+    this.GitPropertiesGenerator.generateGitProperties()
+  }
+
+  object GitPropertiesGenerator {
+    def generateGitProperties(): Unit = {
```

Comment:
pretty cool. Works on Windows if you have `git` available.

Reply:
Added a note in the Read Me and Config about this. Thanks for reminding me